backend/src/app/services/parcelDescriptions/parcelDescriptions.service.ts

```diff
@@ -236,7 +242,106 @@ export class ParcelDescriptionsService {
+            : null;
+        const crownLandsFileNo =
+          parcelDescription.descriptionType === 'Crown Land File Number'
+            ? parcelDescription.idPinNumber
+            : null;
+        return {
+          pid: pid,
+          pin: pin,
+          crownLandsFileNo: crownLandsFileNo,
+          dateNoted: parcelDescription.dateNoted,
+          srAction: parcelDescription.srAction,
+          userAction: parcelDescription.userAction,
+          whoUpdated: userInfo?.givenName,
+          whenUpdated: now,
+          whoCreated: userInfo?.givenName,
+          whenCreated: now,
+          pidStatusCd: 'N', // TODO: Determine what this is and set appropriately.
+        } as Subdivisions;
+      },
+    );
+
+    // Insert the new subdivisions into the database.
+    let insertResult: InsertResult;
+    try {
+      // There is a bug with TypeORM where it will only return columns for the
+      // first parent entity of an entity that uses inheritence. Here I'm using
+      // the querybuilder to work around it.
+      insertResult = await this.subdivisionsRepository
+        .createQueryBuilder()
+        .insert()
+        .into(Subdivisions)
+        .values(subdivisions)
+        .returning('*')
+        .execute();
+    } catch (error) {
+      this.sitesLogger.error(
+        'Exception occured in parcelDescriptionService.addParcelDescriptionsForSite() end',
+        JSON.stringify(error),
+      );
+      throw error;
+    }
+    const newSubdivisions = insertResult.generatedMaps as Subdivisions[];
+
+    // Use the id from the newly created subdivisions to create the related
+    // siteSubdivisions.
+    const siteSubdivisions: SiteSubdivisions[] = newSubdivisions.map(
+      (newSubdivision) => {
+        return {
+          siteId: siteId,
+          subdivId: newSubdivision.id,
+          dateNoted: newSubdivision.dateNoted,
+          userAction: newSubdivision.userAction,
+          srAction: newSubdivision.userAction,
+          whoUpdated: newSubdivision.whoUpdated,
+          whenUpdated: newSubdivision.whenUpdated,
+          whoCreated: newSubdivision.whoCreated,
+          whenCreated: newSubdivision.whenCreated,
+          initialIndicator: 'N', // TODO: Determine what this is an set appropriately.
+          sendToSr: newSubdivision.srAction === 'pending' ? 'Y' : 'N', // TODO: Validate that this is the correct mapping (I don't think we'll actually use this column).
+        } as SiteSubdivisions;
+      },
+    );
+
+    // Commit the new siteSubdivisions to the database.
+    try {
+      await this.siteSubdivisionsRepository.insert(siteSubdivisions);
+    } catch (error) {
+      // This code is called within a transaction in the site service, so the
+      // previous insert should be rolled back if there is a failure here.
+      this.sitesLogger.error(
+        'Exception occured in parcelDescriptionService.addParcelDescriptionsForSite() end',
+        JSON.stringify(error),
+      );
+      throw error;
```

Comment:
if possible can we have a proper formatted Exception here?  for reference disclosure service